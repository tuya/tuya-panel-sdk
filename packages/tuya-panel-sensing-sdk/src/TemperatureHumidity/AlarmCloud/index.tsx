import React, { useRef, useEffect, FC } from 'react';
import { View, Text, StyleSheet, Animated, StyleProp, ViewStyle, TextStyle } from 'react-native';
import { StepperProps, SwitchButton, SwitchButtonProps } from 'tuya-panel-kit';
import { cx } from '../../utils';
import { MultiSlider, Stepper } from '../../components';
import { MultiSliderProps } from '../../components/MultiSlider';

interface AlarmCloudProps {
  title: string;
  minTitle?: string;
  maxTitle?: string;
  showStepComp?: boolean;
  stepMixProps?: StepperProps;
  stepMaxProps?: StepperProps;
  titleStyle?: StyleProp<TextStyle>;
  switchButtonProps?: SwitchButtonProps;
  multiSliderProps?: MultiSliderProps;
  topStyle?: StyleProp<ViewStyle>;
  bottomStyle?: StyleProp<ViewStyle>;
  style?: StyleProp<ViewStyle>;
}

const AlarmCloud: FC<AlarmCloudProps> = props => {
  const {
    title = '湿度告警',
    switchButtonProps,
    multiSliderProps,
    style,
    topStyle,
    bottomStyle,
    titleStyle,
    showStepComp,
    stepMaxProps,
    stepMixProps,
    minTitle = '最小值',
    maxTitle = '最大值',
  } = props;

  const fadeAnim1 = useRef(new Animated.Value(1)).current;

  const fadeOpacity1 = useRef(new Animated.Value(0)).current;

  useEffect(() => {
    if (switchButtonProps?.value) {
      Animated.timing(fadeAnim1, {
        toValue: 1000,
        duration: 1000,
        useNativeDriver: false,
      }).start();
      Animated.timing(fadeOpacity1, {
        toValue: 1,
        duration: 1000,
        useNativeDriver: false,
      }).start();
    } else {
      fadeAnim1.setValue(1);
      fadeOpacity1.setValue(0);
    }
  }, [switchButtonProps?.value]);

  return (
    <View style={[styles.container, style]}>
      <View style={[topStyle, styles.top]}>
        <Text style={titleStyle} numberOfLines={1}>
          {title}
        </Text>
        <SwitchButton {...switchButtonProps!} />
      </View>
      <Animated.View
        style={[
          styles.bottom,
          {
            maxHeight: fadeAnim1,
            opacity: fadeOpacity1,
          },
          bottomStyle,
        ]}
      >
        {switchButtonProps?.value ? (
          <>
            <MultiSlider
              min={0}
              max={100}
              step={1}
              enableLabel
              enabledTwo
              snapped={false}
              labelTextColor="#000000"
              markerOffsetY={2}
              sliderLength={cx(303)}
              values={[0, 100]}
              {...multiSliderProps}
            />
            {showStepComp ? (
              <>
                <View style={styles.stepComp}>
                  <View>
                    <Text style={styles.minTitle}>{minTitle}</Text>
                    <Stepper
                      min={0}
                      max={100}
                      ellipseIconColor="#7F7F7F"
                      placeholderTextColor="#FFFFFF"
                      value={20}
                      stepValue={1}
                      editable={false}
                      {...stepMixProps}
                    />
                  </View>
                  <View>
                    <Text style={styles.maxTitle}>{maxTitle}</Text>
                    <Stepper
                      min={0}
                      max={100}
                      ellipseIconColor="#7F7F7F"
                      placeholderTextColor="#FFFFFF"
                      value={20}
                      stepValue={1}
                      editable={false}
                      {...stepMaxProps}
                    />
                  </View>
                </View>
              </>
            ) : null}
          </>
        ) : null}
      </Animated.View>
    </View>
  );
};

AlarmCloud.defaultProps = {
  switchButtonProps: {
    value: false,
    onValueChange: () => undefined,
  },
  multiSliderProps: {
    min: 0,
    max: 100,
    step: 1,
    enabledTwo: true,
    enableLabel: true,
    snapped: false,
    labelTextColor: '#000000',
    markerOffsetY: 2,
    sliderLength: cx(303),
    values: [0, 100],
  },
};

const styles = StyleSheet.create({
  bottom: {
    flexDirection: 'column',
    marginTop: 18,
  },
  container: {
    backgroundColor: '#ffffff',
  },
  maxTitle: {
    color: '#ffffff',
    marginBottom: 5,
  },
  minTitle: {
    color: '#ffffff',
    marginBottom: 5,
  },
  stepComp: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 10,
  },
  top: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
});

export default AlarmCloud;
