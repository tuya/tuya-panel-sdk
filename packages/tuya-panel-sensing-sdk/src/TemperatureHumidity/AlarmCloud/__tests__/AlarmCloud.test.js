/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow, mount } from 'enzyme';
import AlarmCloud from '../index';

global.document = document;
global.window = window;

describe('MultiSlider', () => {
  it('basic demo', () => {
    const wrapper = shallow(<AlarmCloud />);
    expect(wrapper).toMatchSnapshot();
  });

  it('basic demo1', () => {
    const wrapper = shallow(<AlarmCloud min={0} max={100} />);
    expect(wrapper).toMatchSnapshot();
  });
});
