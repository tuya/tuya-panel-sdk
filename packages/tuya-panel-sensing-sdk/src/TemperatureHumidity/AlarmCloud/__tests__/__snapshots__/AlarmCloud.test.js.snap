// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSlider basic demo 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      numberOfLines={1}
    >
      湿度告警
    </Text>
    <ThemedSwitchButton
      onValueChange={[Function]}
      theme={null}
      value={false}
    />
  </View>
  <AnimatedComponent
    style={
      Array [
        Object {
          "flexDirection": "column",
          "marginTop": 18,
        },
        Object {
          "maxHeight": 1,
          "opacity": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`MultiSlider basic demo1 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      numberOfLines={1}
    >
      湿度告警
    </Text>
    <ThemedSwitchButton
      onValueChange={[Function]}
      theme={null}
      value={false}
    />
  </View>
  <AnimatedComponent
    style={
      Array [
        Object {
          "flexDirection": "column",
          "marginTop": 18,
        },
        Object {
          "maxHeight": 1,
          "opacity": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;
